### PR TITLE
Admob implementation

### PR DESCRIPTION
**Pull request by RagnarR**
* Added new Custom View.
* Added AdMob adapter for NativeAd.
* Added AdMob adapter for InterstitialAd.

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -1,6 +1,10 @@
 machine:
   timezone: Europe/Berlin
 
+dependencies:
+  pre:
+    - echo y | sudo /usr/local/android-sdk-linux/tools/android update sdk --no-ui --all --filter extra-android-m2repository,extra-android-support,extra-google-google_play_services,extra-google-m2repository
```
Comment (cerberillo) [inline circle.yml]: what is this for?

File: mediation.demo/src/main/java/net/pubnative/mediation/adapter/model/AdMobNativeAppInstallAdModel.java [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+    public static final String TAG = AdMobNativeAppInstallAdModel.class.getSimpleName();
+
+    protected NativeAppInstallAd mNativeAd;
+
+    public AdMobNativeAppInstallAdModel(NativeAppInstallAd nativeAd) {
+
+        if (nativeAd != null) {
+            mNativeAd = nativeAd;
+        }
```
Comment (cerberillo) [inline mediation.demo/src/main/java/net/pubnative/mediation/adapter/model/AdMobNativeAppInstallAdModel.java]: You should assign independently of the value

File: mediation.demo/src/main/java/net/pubnative/mediation/adapter/model/AdMobNativeAppInstallAdModel.java [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+    }
+
+    @Override
+    public String getBannerUrl() {
+
+        Log.v(TAG, "getBannerUrl");
+
+        String result = null;
+        if (mNativeAd != null && mNativeAd.getImages() != null) {
+            if (mNativeAd.getImages().size() > 0) {
```
Comment (cerberillo) [inline mediation.demo/src/main/java/net/pubnative/mediation/adapter/model/AdMobNativeAppInstallAdModel.java]: can't we include tis inside the if sentence?

File: circle.yml (hunk 1 above)
Comment (RagnarR) [inline circle.yml]: For AdMob plugin - it uses last play services. Without it, we will get crash during the build

File: mediation.demo/src/main/java/net/pubnative/mediation/adapter/model/AdMobNativeAppInstallAdModel.java (hunk 3 above)
Comment (RagnarR) [inline mediation.demo/src/main/java/net/pubnative/mediation/adapter/model/AdMobNativeAppInstallAdModel.java]: 👍

File: mediation.demo/src/main/java/net/pubnative/mediation/adapter/model/FacebookNativeAdModel.java [hunk 4]
```diff
@@ -139,6 +139,13 @@ public View getAdvertisingDisclosureView(Context context) {
         }
         return result;
     }
+
+    @Override
+    public Object getNativeAd() {
+
+        return mNativeAd;
+    }
```
Comment (cerberillo) [inline mediation.demo/src/main/java/net/pubnative/mediation/adapter/model/FacebookNativeAdModel.java]: One question, can't the widget extract the inner item using a package protected method instead of exposing this publicly?

File: mediation.demo/src/main/java/net/pubnative/mediation/adapter/model/AdMobNativeAppInstallAdModel.java (hunk 2 above)
Comment (RagnarR) [inline mediation.demo/src/main/java/net/pubnative/mediation/adapter/model/AdMobNativeAppInstallAdModel.java]: Not sure that I understand it correctly. You mean assign in any case - without checking?

File: mediation.demo/src/main/java/net/pubnative/mediation/adapter/network/AdMobNetworkRequestAdapter.java [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+import net.pubnative.mediation.request.model.PubnativeAdTargetingModel;
+
+import java.util.Calendar;
+import java.util.GregorianCalendar;
+import java.util.Map;
+
+public class AdMobNetworkRequestAdapter extends PubnativeNetworkRequestAdapter implements NativeAppInstallAd.OnAppInstallAdLoadedListener {
+
+    public static final String    TAG       = AdMobNetworkRequestAdapter.class.getSimpleName();
+    protected static final String ADMOB_UNIT_ID   = "unit_id";
```
Comment (cerberillo) [inline mediation.demo/src/main/java/net/pubnative/mediation/adapter/network/AdMobNetworkRequestAdapter.java]: please align

File: mediation.demo/src/main/java/net/pubnative/mediation/adapter/network/AdMobNetworkRequestAdapter.java [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+
+        Log.v(TAG, "request");
+        if (context != null && mData != null) {
+            String placementId = (String) mData.get(ADMOB_UNIT_ID);
+            if (TextUtils.isEmpty(placementId)) {
+                invokeFailed(PubnativeException.ADAPTER_ILLEGAL_ARGUMENTS);
+            } else {
+                createRequest(context, placementId);
+            }
+        }
```
Comment (cerberillo) [inline mediation.demo/src/main/java/net/pubnative/mediation/adapter/network/AdMobNetworkRequestAdapter.java]: else fail with `ADAPTER_ILLEGAL_ARGUMENTS `

File: mediation.demo/src/main/java/net/pubnative/mediation/adapter/network/AdMobNetworkRequestAdapter.java [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+    //==============================================================================================
+
+    @Override
+    protected void request(Context context) {
+
+        Log.v(TAG, "request");
+        if (context != null && mData != null) {
+            String placementId = (String) mData.get(ADMOB_UNIT_ID);
+            if (TextUtils.isEmpty(placementId)) {
+                invokeFailed(PubnativeException.ADAPTER_ILLEGAL_ARGUMENTS);
```
Comment (cerberillo) [inline mediation.demo/src/main/java/net/pubnative/mediation/adapter/network/AdMobNetworkRequestAdapter.java]: this is an `adapter_missing_arguments`

File: mediation.demo/src/main/java/net/pubnative/mediation/adapter/widget/PubnativeNativeAdView.java [hunk 8]
```diff
@@ -0,0 +1,203 @@
...
+    // PubnativeNativeAdView methods
+    //==============================================================================================
+    // Initializing
+    //----------------------------------------------------------------------------------------------
+
+    protected void initLayout(Context context) {
+
+        Log.v(TAG, "initLayout");
+        mContext = context;
+    }
```
Comment (cerberillo) [inline mediation.demo/src/main/java/net/pubnative/mediation/adapter/widget/PubnativeNativeAdView.java]: Can't we capture this in the updateAdView?

File: mediation.demo/src/main/java/net/pubnative/mediation/adapter/widget/PubnativeNativeAdView.java [hunk 9]
```diff
@@ -0,0 +1,203 @@
...
+    public void updateAdView(PubnativeAdModel model) {
+
+        Log.v(TAG, "updateAdView");
+        // Ad content
+        setModel(model);
+    }
+
+    protected void setModel(PubnativeAdModel model) {
+
+        mAdModel = model;
```
Comment (cerberillo) [inline mediation.demo/src/main/java/net/pubnative/mediation/adapter/widget/PubnativeNativeAdView.java]: missing log

File: mediation.demo/src/main/java/net/pubnative/mediation/adapter/model/FacebookNativeAdModel.java (hunk 4 above)
Comment (RagnarR) [inline mediation.demo/src/main/java/net/pubnative/mediation/adapter/model/FacebookNativeAdModel.java]: Sure, it is my mistake using pre generated template

File: mediation.demo/src/main/java/net/pubnative/mediation/adapter/widget/PubnativeNativeAdView.java [hunk 10]
```diff
@@ -0,0 +1,203 @@
...
+    /**
+     * Populate data in view
+     *
+     * @param model Native Ad model
+     */
+    public void updateAdView(PubnativeAdModel model) {
+
+        Log.v(TAG, "updateAdView");
+        // Ad content
+        setModel(model);
```
Comment (cerberillo) [inline mediation.demo/src/main/java/net/pubnative/mediation/adapter/widget/PubnativeNativeAdView.java]: this method could be removed and use setModel instead

File: mediation.demo/src/main/java/net/pubnative/mediation/adapter/widget/PubnativeNativeAdView.java [hunk 11]
```diff
@@ -0,0 +1,203 @@
...
+    }
+
+    protected void setModel(PubnativeAdModel model) {
+
+        mAdModel = model;
+
+        populateAdView();
```
Comment (cerberillo) [inline mediation.demo/src/main/java/net/pubnative/mediation/adapter/widget/PubnativeNativeAdView.java]: you should check for null model before invoking this method, otherwise this could crash

File: mediation.demo/src/main/java/net/pubnative/mediation/adapter/widget/PubnativeNativeAdView.java [hunk 12]
```diff
@@ -0,0 +1,203 @@
...
+        View sponsorView = mAdModel.getAdvertisingDisclosureView(mContext);
+        if (sponsorView != null) {
+            mAdDisclosure.addView(sponsorView);
+        }
+        if (!TextUtils.isEmpty(mAdModel.getCallToAction())) {
+            mCallToAction.setVisibility(VISIBLE);
+            mCallToAction.setText(mAdModel.getCallToAction());
+        } else {
+            mCallToAction.setVisibility(GONE);
+        }
```
Comment (cerberillo) [inline mediation.demo/src/main/java/net/pubnative/mediation/adapter/widget/PubnativeNativeAdView.java]: This method is generally incorrect, because you presume here only 1 possible behaviour and It should contain a lot more checks

In example for the title:
```java
if(mTitle != null) {
  mTitle.setText(mAdModel.getTitle());
}
```

File: mediation.demo/src/main/java/net/pubnative/mediation/adapter/widget/PubnativeNativeAdView.java [hunk 13]
```diff
@@ -0,0 +1,203 @@
...
+        View sponsorView = mAdModel.getAdvertisingDisclosureView(mContext);
+        if (sponsorView != null) {
+            mAdDisclosure.addView(sponsorView);
+        }
+        if (!TextUtils.isEmpty(mAdModel.getCallToAction())) {
+            mCallToAction.setVisibility(VISIBLE);
+            mCallToAction.setText(mAdModel.getCallToAction());
+        } else {
+            mCallToAction.setVisibility(GONE);
```
Comment (cerberillo) [inline mediation.demo/src/main/java/net/pubnative/mediation/adapter/widget/PubnativeNativeAdView.java]: why are you handling visibility of elements? this shouldn't be accomplished by you in any case

File: mediation.demo/src/main/java/net/pubnative/mediation/adapter/widget/PubnativeNativeAdView.java [hunk 14]
```diff
@@ -0,0 +1,203 @@
...
+
+    public void setStoreView(TextView view) {
+
+        mStore = view;
+    }
+
+    public void setAdDisclosure(ViewGroup view) {
+
+        mAdDisclosure = view;
+    }
```
Comment (cerberillo) [inline mediation.demo/src/main/java/net/pubnative/mediation/adapter/widget/PubnativeNativeAdView.java]: please add javadoc to all public methods

File: mediation/google-services.json [hunk 15]
```diff
@@ -0,0 +1,42 @@
+{
```
Comment (cerberillo) [inline mediation/google-services.json]: why is this here? This should be included only in the client app part

File: mediation/src/main/java/net/pubnative/mediation/config/model/PubnativePriorityRuleModel.java [hunk 16]
```diff
@@ -37,4 +37,15 @@
...
+
+    @Override
+    public String toString() {
+
+        return "PubnativePriorityRuleModel{" +
+               "id=" + id +
+               ", network_code='" + network_code + '\'' +
+               ", params=" + params +
+               ", segment_ids=" + segment_ids +
+               '}';
```
Comment (cerberillo) [inline mediation/src/main/java/net/pubnative/mediation/config/model/PubnativePriorityRuleModel.java]: What is this for?

File: mediation/src/test/java/net/pubnative/mediation/insights/model/PubnativeInsightDataModelTest.java [hunk 17]
```diff
@@ -88,6 +88,30 @@ public void addInterestWithDifferentValues() {
...
+        dataModelSpy.reset();
+
+        // network as empty string
+        dataModelSpy.addUnreachableNetwork("");
+        assertThat(dataModelSpy.unreachable_networks).isNull();
+
+        // network as null
+        dataModelSpy.addUnreachableNetwork(null);
+        assertThat(dataModelSpy.unreachable_networks).isNull();
+    }
```
Comment (cerberillo) [inline mediation/src/test/java/net/pubnative/mediation/insights/model/PubnativeInsightDataModelTest.java]: as attempted networks, `addUnreachableNetworkWithDifferentValues ` should be split in different methods

File: mediation.demo/src/main/java/net/pubnative/mediation/adapter/network/AdMobNetworkRequestAdapter.java (hunk 5 above)
Comment (RagnarR) [inline mediation.demo/src/main/java/net/pubnative/mediation/adapter/network/AdMobNetworkRequestAdapter.java]: 👍

File: mediation.demo/src/main/java/net/pubnative/mediation/adapter/network/AdMobNetworkRequestAdapter.java (hunk 6 above)
Comment (RagnarR) [inline mediation.demo/src/main/java/net/pubnative/mediation/adapter/network/AdMobNetworkRequestAdapter.java]: Did you mean for first if statement?

File: mediation.demo/src/main/java/net/pubnative/mediation/adapter/network/AdMobNetworkRequestAdapter.java (hunk 7 above)
Comment (RagnarR) [inline mediation.demo/src/main/java/net/pubnative/mediation/adapter/network/AdMobNetworkRequestAdapter.java]: 👍

File: mediation.demo/src/main/java/net/pubnative/mediation/adapter/widget/PubnativeNativeAdView.java (hunk 13 above)
Comment (RagnarR) [inline mediation.demo/src/main/java/net/pubnative/mediation/adapter/widget/PubnativeNativeAdView.java]: make sense

File: mediation/src/main/java/net/pubnative/mediation/config/model/PubnativePriorityRuleModel.java (hunk 16 above)
Comment (RagnarR) [inline mediation/src/main/java/net/pubnative/mediation/config/model/PubnativePriorityRuleModel.java]: Forgot delete after debugging

File: mediation.demo/src/main/java/net/pubnative/mediation/adapter/widget/PubnativeNativeAdView.java (hunk 14 above)
Comment (RagnarR) [inline mediation.demo/src/main/java/net/pubnative/mediation/adapter/widget/PubnativeNativeAdView.java]: I thought that it very obvious methods and comments not necessary.

File: mediation/google-services.json (hunk 15 above)
Comment (RagnarR) [inline mediation/google-services.json]: ok, I agree.

File: mediation/src/test/java/net/pubnative/mediation/insights/model/PubnativeInsightDataModelTest.java (hunk 17 above)
Comment (RagnarR) [inline mediation/src/test/java/net/pubnative/mediation/insights/model/PubnativeInsightDataModelTest.java]: It was merged from old tests, but ok - I will fix it :)

File: mediation.demo/src/main/java/net/pubnative/mediation/adapter/model/FacebookNativeAdModel.java (hunk 4 above)
Comment (RagnarR) [inline mediation.demo/src/main/java/net/pubnative/mediation/adapter/model/FacebookNativeAdModel.java]: I tried, but this is a better solution, what I found. And if we will set this method as protected - we will not have possibility to work with it outside the class.

File: mediation.demo/src/main/java/net/pubnative/mediation/adapter/widget/PubnativeNativeAdView.java (hunk 8 above)
Comment (RagnarR) [inline mediation.demo/src/main/java/net/pubnative/mediation/adapter/widget/PubnativeNativeAdView.java]: No, because we capturing context not only after programmatically creation but also after using this view inside the XML. I can move it in each constructor.

File: mediation.demo/src/main/java/net/pubnative/mediation/adapter/network/AdMobNetworkRequestAdapter.java [hunk 18]
```diff
@@ -0,0 +1,118 @@
...
+    //==============================================================================================
+    // PubnativeNetworkAdapter methods
+    //==============================================================================================
+
+    @Override
+    protected void request(Context context) {
+
+        Log.v(TAG, "request");
+        if (context != null && mData != null) {
+            String placementId = (String) mData.get(ADMOB_UNIT_ID);
```
Comment (cerberillo) [inline mediation.demo/src/main/java/net/pubnative/mediation/adapter/network/AdMobNetworkRequestAdapter.java]: the name of the variable should be unitId instead

File: mediation.demo/src/main/java/net/pubnative/mediation/adapter/widget/PubnativeNativeAdView.java [hunk 19]
```diff
@@ -0,0 +1,203 @@
...
+        super(context, attrs, defStyleAttr);
+        initLayout(context);
+    }
+
+    @Override
+    public boolean onInterceptTouchEvent(MotionEvent ev) {
+
+        Log.v(TAG, "onInterceptTouchEvent");
+        onTouchEvent(ev);
+        return false;
```
Comment (cerberillo) [inline mediation.demo/src/main/java/net/pubnative/mediation/adapter/widget/PubnativeNativeAdView.java]: how is this working?

File: mediation.demo/src/main/java/net/pubnative/mediation/adapter/network/AdMobNetworkRequestAdapter.java (hunk 6 above)
Comment (cerberillo) [inline mediation.demo/src/main/java/net/pubnative/mediation/adapter/network/AdMobNetworkRequestAdapter.java]:
```java
if(context == null || data == null){
  invokeFailed(PubnativeException.ADAPTER_ILLEGAL_ARGUMENTS);
} else {
  String placementId = (String) mData.get(ADMOB_UNIT_ID);
  if (TextUtils.isEmpty(placementId)) {
    invokeFailed(PubnativeException.ADAPTER_MISSING_ARGUMENTS);
  } else {
    createRequest(context, placementId);
  }
}
```

File: mediation.demo/src/main/java/net/pubnative/mediation/adapter/widget/PubnativeNativeAdView.java (hunk 8 above)
Comment (cerberillo) [inline mediation.demo/src/main/java/net/pubnative/mediation/adapter/widget/PubnativeNativeAdView.java]: it's ok here, but I would get it in the first moment that we really need it, if it's on inflate then it's ok as it is

File: mediation.demo/src/main/java/net/pubnative/mediation/adapter/widget/PubnativeNativeAdView.java (hunk 14 above)
Comment (cerberillo) [inline mediation.demo/src/main/java/net/pubnative/mediation/adapter/widget/PubnativeNativeAdView.java]: we need the javadoc for bintray purpose even being obvious

File: mediation.demo/src/main/java/net/pubnative/mediation/adapter/widget/PubnativeNativeAdView.java (hunk 19 above)
Comment (RagnarR) [inline mediation.demo/src/main/java/net/pubnative/mediation/adapter/widget/PubnativeNativeAdView.java]: It passed touch event to parent from all child views. It allows fixing the issue with PubnativeLibraryAdModel.

File: mediation.demo/src/main/java/net/pubnative/mediation/adapter/network/AdMobNetworkRequestAdapter.java (hunk 18 above)
Comment (RagnarR) [inline mediation.demo/src/main/java/net/pubnative/mediation/adapter/network/AdMobNetworkRequestAdapter.java]: Did you mean `ADMOB_UNITID` ?

Comment (RagnarR) [inline mediation.demo/src/main/java/net/pubnative/mediation/adapter/network/AdMobNetworkRequestAdapter.java]: 👍

File: mediation.demo/src/main/java/net/pubnative/mediation/adapter/widget/PubnativeNativeAdView.java (hunk 12 above)
Comment (RagnarR) [inline mediation.demo/src/main/java/net/pubnative/mediation/adapter/widget/PubnativeNativeAdView.java]: 👍

File: mediation.demo/src/main/java/net/pubnative/mediation/adapter/network/AdMobNetworkRequestAdapter.java (hunk 6 above)
Comment (RagnarR) [inline mediation.demo/src/main/java/net/pubnative/mediation/adapter/network/AdMobNetworkRequestAdapter.java]: ok, first one :)

File: mediation.demo/src/main/java/net/pubnative/mediation/adapter/widget/PubnativeNativeAdView.java (hunk 9 above)
Comment (RagnarR) [inline mediation.demo/src/main/java/net/pubnative/mediation/adapter/widget/PubnativeNativeAdView.java]: 👍

File: mediation.demo/src/main/java/net/pubnative/mediation/adapter/widget/PubnativeNativeAdView.java (hunk 10 above)
Comment (RagnarR) [inline mediation.demo/src/main/java/net/pubnative/mediation/adapter/widget/PubnativeNativeAdView.java]: 👍

File: mediation.demo/src/main/java/net/pubnative/mediation/adapter/widget/PubnativeNativeAdView.java (hunk 14 above)
Comment (RagnarR) [inline mediation.demo/src/main/java/net/pubnative/mediation/adapter/widget/PubnativeNativeAdView.java]: ok, got it

File: mediation.demo/src/main/java/net/pubnative/mediation/adapter/widget/PubnativeNativeAdView.java (hunk 11 above)
Comment (RagnarR) [inline mediation.demo/src/main/java/net/pubnative/mediation/adapter/widget/PubnativeNativeAdView.java]: 👍

Comment (RagnarR) [inline mediation.demo/src/main/java/net/pubnative/mediation/adapter/widget/PubnativeNativeAdView.java]: @cerberillo btw, how do you looking for putting in log some warning message if model will be null?
something like `Log.w(TAG, "setModel: model is null");`

File: mediation.demo/src/main/java/net/pubnative/mediation/adapter/widget/PubnativeNativeAdView.java (hunk 14 above)
Comment (RagnarR) [inline mediation.demo/src/main/java/net/pubnative/mediation/adapter/widget/PubnativeNativeAdView.java]: 👍

File: mediation.demo/src/main/java/net/pubnative/mediation/adapter/widget/PubnativeNativeAdView.java (hunk 12 above)
Comment (RagnarR) [inline mediation.demo/src/main/java/net/pubnative/mediation/adapter/widget/PubnativeNativeAdView.java]: @cerberillo do we need to add checks for `populateAppInstallAdView` method? Or leave this checks for AdMob?

File: mediation/google-services.json (hunk 15 above)
Comment (RagnarR) [inline mediation/google-services.json]: btw, I not sure, that app will build without it

File: mediation/src/test/java/net/pubnative/mediation/insights/model/PubnativeInsightDataModelTest.java (hunk 17 above)
Comment (cerberillo) [inline mediation/src/test/java/net/pubnative/mediation/insights/model/PubnativeInsightDataModelTest.java]: Did you split this?

File: mediation.demo/build.gradle [hunk 20]
```diff
@@ -9,9 +9,10 @@ buildscript {
         maven { url 'https://maven.fabric.io/public' }
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:2.1.0'
+        classpath 'com.android.tools.build:gradle:2.1.2'
```
Comment (cerberillo) [inline mediation.demo/build.gradle]: We shouldn't change gradle version this easily, please leave this until agreed with everybody that we will change gradle version

File: mediation.demo/src/main/java/net/pubnative/mediation/demo/AdViewHolder.java [hunk 21]
```diff
@@ -125,12 +127,17 @@ public void renderAd() {
             mRating.setVisibility(View.VISIBLE);
             Picasso.with(mContext).load(model.getIconUrl()).into(mIcon);
             Picasso.with(mContext).load(model.getBannerUrl()).into(mBanner);
-            View sponsorView = model.getAdvertisingDisclosureView(mContext);
+            View sponsorView = model.getAdvertisingDisclosureView(this.mContext);
```
Comment (cerberillo) [inline mediation.demo/src/main/java/net/pubnative/mediation/demo/AdViewHolder.java]: remove `this.` please

File: mediation/src/test/java/net/pubnative/mediation/insights/model/PubnativeInsightDataModelTest.java [hunk 22]
```diff
@@ -88,6 +88,44 @@ public void addInterestWithDifferentValues() {
     }
 
     @Test
+    public void addUnreachableNetwork_withValidValue_isNotNull() {
+        PubnativeInsightDataModel dataModelSpy = spy(PubnativeInsightDataModel.class);
+
+        // the list is null at the beginning
+        assertThat(dataModelSpy.unreachable_networks).isNull();
```
Comment (cerberillo) [inline mediation/src/test/java/net/pubnative/mediation/insights/model/PubnativeInsightDataModelTest.java]: We don't need to do this check at the begining

File: mediation/src/test/java/net/pubnative/mediation/insights/model/PubnativeInsightDataModelTest.java [hunk 23]
```diff
@@ -88,6 +88,44 @@ public void addInterestWithDifferentValues() {
...
+
+        // valid string
+        dataModelSpy.addUnreachableNetwork(validString);
```
Comment (cerberillo) [inline mediation/src/test/java/net/pubnative/mediation/insights/model/PubnativeInsightDataModelTest.java]: Let's use validString here directly instead of using a static String

File: mediation.demo/src/main/java/net/pubnative/mediation/demo/AdViewHolder.java (hunk 21 above)
Comment (RagnarR) [inline mediation.demo/src/main/java/net/pubnative/mediation/demo/AdViewHolder.java]: 👍

File: mediation/src/test/java/net/pubnative/mediation/insights/model/PubnativeInsightDataModelTest.java (hunk 22 above)
Comment (RagnarR) [inline mediation/src/test/java/net/pubnative/mediation/insights/model/PubnativeInsightDataModelTest.java]: Ok, will remove

File: mediation/src/test/java/net/pubnative/mediation/insights/model/PubnativeInsightDataModelTest.java (hunk 23 above)
Comment (RagnarR) [inline mediation/src/test/java/net/pubnative/mediation/insights/model/PubnativeInsightDataModelTest.java]: ok

File: mediation/src/test/java/net/pubnative/mediation/insights/model/PubnativeInsightDataModelTest.java (hunk 17 above)
Comment (RagnarR) [inline mediation/src/test/java/net/pubnative/mediation/insights/model/PubnativeInsightDataModelTest.java]: yes

File: mediation.demo/build.gradle [hunk 24]
```diff
@@ -51,6 +52,9 @@ dependencies {
     // FLURRY
     compile 'com.flurry.android:analytics:6.2.0'
     compile 'com.flurry.android:ads:6.2.0'
+    //  ADMOB
+    compile 'com.google.firebase:firebase-core:9.0.0'
```
Comment (cerberillo) [inline mediation.demo/build.gradle]: Do we need the core for something? I think that we can work just with `ads` repository. In fact I would say that `ads` repository includes `core`

Comment (RagnarR) [inline mediation.demo/build.gradle]: If I remember I leave `core` for flurry native Ad.

**Comment by cerberillo** [comment]
Once the core removal is tested that works, this is ready to go 👍
